packages/lib/src/Login.wc.svelte

```diff
@@ -1,46 +1,70 @@
 <svelte:options customElement="sesamy-login" />
 
 <script lang="ts">
+  import Base from "./Base.svelte";
+  import type { Api } from "./api";
+
   let loading = false;
   let loggedIn = false;
   let userAvatar = "https://via.placeholder.com/50"; // Placeholder avatar image
 
-  // Mock function to simulate login process
-  const login = async () => {
+  // Function to handle login
+  const login = async (api: Api) => {
     loading = true;
-    await new Promise((resolve) => setTimeout(resolve, 2000)); // Simulate loading
-    loading = false;
-    loggedIn = true;
+    try {
+      // console.log("Logging in...");
+      await api.auth.loginWithRedirect();
+    } catch (error) {
+      console.error("Login failed:", error);
+    } finally {
+      loading = false;
+    }
   };
 
-  // Mock function to simulate logout process
-  const logout = () => {
-    loggedIn = false;
+  // Function to handle logout
+  const logout = async (api: Api) => {
+    try {
+      await api.auth.logout();
+      loggedIn = false;
+    } catch (error) {
+      console.error("Logout failed:", error);
+    }
   };
 
-  // Mock function to check if user is logged in (initially false)
-  const checkLoggedIn = () => {
-    return loggedIn;
+  // Function to check if user is logged in
+  const checkLoggedIn = async (api: Api) => {
+    try {
+      loggedIn = await api.auth.isAuthenticated();
+    } catch (error) {
+      console.error("Error checking login status:", error);
+    }
   };
-
-  // Simulate login status on component load
-  loggedIn = checkLoggedIn();
 </script>
 
-{#if loading}
-  <!-- Loading State -->
-  <button class="loading-btn" aria-busy="true" aria-label="Logging in">
-    <!-- Pulsating animation for loading state -->
-  </button>
-{:else if loggedIn}
-  <!-- Logged In State -->
-  <button class="avatar-btn" onclick={logout} aria-label="Log out">
-    <img src={userAvatar} alt="User Avatar" />
-  </button>
-{:else}
-  <!-- Logged Out State -->
-  <button class="login-btn" onclick={login} aria-label="Log in"> Login </button>
-{/if}
+<Base let:api>
+  {#await checkLoggedIn(api) then _}
```

Comment:
Let's run for a bit with these `#await`'s.

Another solution is to assume you are not logged in and then render the logged in state if sesamy-js thinks you are.

Can this code ever render "blank" for 5+ seconds for example? We wouldn't want that.